lib/AST/Decl.cpp

```diff
@@ -5967,7 +6000,20 @@ ParamDecl::getDefaultValueStringRepresentation(
   case DefaultArgumentKind::Column: return "#column";
   case DefaultArgumentKind::Function: return "#function";
   case DefaultArgumentKind::DSOHandle: return "#dsohandle";
-  case DefaultArgumentKind::NilLiteral: return "nil";
+  case DefaultArgumentKind::NilLiteral: {
+    // NilLiteral can both mean 'nil' and 'tuple of nils'.
+    if (getInterfaceType()->getAnyNominal() ==
+        getASTContext().getOptionalDecl())
+      return "nil";
+
+    assert(getInterfaceType()->is<TupleType>() &&
+           "non-nil default initialized value must be a tuple");
+
+    llvm::raw_svector_ostream os(scratch);
+    writeTupleOfNils(getInterfaceType()->getAs<TupleType>(), getASTContext(),
```

Comment:
Remove the `is()` assert and use `castTo()` here instead of `getAs()`; it will assert if the cast fails.